src/vcpkg/configuration.cpp

```diff
@@ -191,112 +165,107 @@ namespace
     const RegistriesArrayDeserializer RegistriesArrayDeserializer::instance;
 
     View<StringView> RegistryConfigDeserializer::valid_fields() const
-    {
-        static constexpr StringView t[] = {KIND, BASELINE, PATH, REPO, REFERENCE, NAME, LOCATION};
-        return t;
-    }
-    View<StringView> valid_builtin_fields()
-    {
-        static constexpr StringView t[] = {
-            RegistryConfigDeserializer::KIND,
-            RegistryConfigDeserializer::BASELINE,
-            RegistryDeserializer::PACKAGES,
-        };
-        return t;
-    }
-    View<StringView> valid_filesystem_fields()
     {
         static constexpr StringView t[] = {
-            RegistryConfigDeserializer::KIND,
-            RegistryConfigDeserializer::BASELINE,
-            RegistryConfigDeserializer::PATH,
-            RegistryDeserializer::PACKAGES,
-        };
-        return t;
-    }
-    View<StringView> valid_git_fields()
-    {
-        static constexpr StringView t[] = {
-            RegistryConfigDeserializer::KIND,
-            RegistryConfigDeserializer::BASELINE,
-            RegistryConfigDeserializer::REPO,
-            RegistryConfigDeserializer::REFERENCE,
-            RegistryDeserializer::PACKAGES,
-        };
-        return t;
-    }
-    View<StringView> valid_artifact_fields()
-    {
-        static constexpr StringView t[] = {
-            RegistryConfigDeserializer::KIND,
-            RegistryConfigDeserializer::NAME,
-            RegistryConfigDeserializer::LOCATION,
-        };
+            JsonIdKind, JsonIdBaseline, JsonIdPath, JsonIdRepository, JsonIdReference, JsonIdName, JsonIdLocation};
         return t;
     }
+    static constexpr StringView valid_builtin_fields[] = {
+        JsonIdKind,
+        JsonIdBaseline,
+        JsonIdPackages,
+    };
+
+    static constexpr StringView valid_filesystem_fields[] = {
+        JsonIdKind,
+        JsonIdBaseline,
+        JsonIdPath,
+        JsonIdPackages,
+    };
+
+    static constexpr StringView valid_git_fields[] = {
+        JsonIdKind,
+        JsonIdBaseline,
+        JsonIdRepository,
+        JsonIdReference,
+        JsonIdPackages,
+    };
+
+    static constexpr StringView valid_artifact_fields[] = {
+        JsonIdKind,
+        JsonIdName,
+        JsonIdLocation,
+    };
 
     Optional<RegistryConfig> RegistryConfigDeserializer::visit_null(Json::Reader&) const { return RegistryConfig(); }
 
     Optional<RegistryConfig> RegistryConfigDeserializer::visit_object(Json::Reader& r, const Json::Object& obj) const
     {
         RegistryConfig res;
         auto& kind = res.kind.emplace();
-        r.required_object_field(type_name(), obj, KIND, kind, RegistryImplementationKindDeserializer::instance);
+        r.required_object_field(type_name(), obj, JsonIdKind, kind, RegistryImplementationKindDeserializer::instance);
 
-        if (kind == KIND_BUILTIN)
+        if (kind == JsonIdBuiltin)
```

Comment:
Consideration: should these semantically be `JsonValueBuiltin` instead, since they aren't keys? (As-is also LGTM)

Reply:
I originally had `JsonFieldXxx` but this convinced me to do `JsonId` instead because they were sometimes values.